src/qibolab/designs/basic.py

```diff
@@ -1,8 +1,61 @@
 from dataclasses import dataclass, field
+from typing import Any, List, Optional
 
 from qibolab.instruments.abstract import AbstractInstrument
 
 
+@dataclass
+class Channel:
+    """Representation of physical wire connection (channel).
+
+    Name is used as a unique identifier for channels.
+    Channel objects are instantiated by :class:`qibolab.platforms.platform.Platform`,
+    but their attributes are modified and used by instrument designs.
+
+    Args:
+        name (str): Name of the channel as given in the platform runcard.
+
+    Attributes:
+        ports (list): List of tuples (controller (`str`), port (`int`))
+            specifying the QM (I, Q) ports that the channel is connected.
+        qubits (list): List of tuples (:class:`qibolab.platforms.utils.Qubit`, str)
+            for the qubit connected to this channel and the role of the channel.
+        Optional arguments holding local oscillators and related parameters.
+        These are relevant only for mixer-based insturment designs.
+    """
+
+    name: str
+
+    qubit: Optional["Qubit"] = field(default=None, init=False, repr=False)
+    ports: List[tuple] = field(default_factory=list, init=False)
+    local_oscillator: Any = field(default=None, init=False)
+    lo_frequency: float = field(default=0, init=False)
+    lo_power: float = field(default=0, init=False)
+    _offset: Optional[float] = field(default=None, init=False)
+    _filter: Optional[dict] = field(default=None, init=False)
```

Comment:
Maybe there is a misunderstanding on the role of `init` kwarg (and maybe not): since you are always setting a default you can always omit that parameter from the constructor.

The only difference is that with `init=False` you can't set it at all. So `init=False` is preventing users to set, e.g., the `lo_power` parameter during construction, but they can set it immediately after:
- wrong
    ```py
    ch = Channel("ciao", lo_power=1.0)
    ```
- still correct
    ```py
    ch = Channel("ciao")
    ch.lo_power=1.0
    ```

Reply:
You are right, thanks.